### PR TITLE
Add AnalyzeCommand_DeterminismTest and good repro files

### PR DESCRIPTION
**Pull request by shaopeng-gh**
### Update: Now we have Coyote Test ready, there is no need for `RepeatAttribute` I have removed it, the test itself is valid and the repro files are good repro files, I think we can check it in.

The tests files are existing test files in the solution, I did experiments to found the best one that repro this issue. This files are even better repro than the user reported ones.

--------------
during add test for that Determinism issue,
I propose this way before implement Coyote.
Coyote need to mock things that Coyote not in control e.g. file system, that probably needs code changes to tests.
For that Determinism issue actually can have a test that can repro every time without taking much time.
**I added a `RepeatAttribute` we can add to the tests those fail randomly.**
The Attribute will require the test to be `[Theory]` but I have tested it does not slow down the test, the one that slows down is the `[InlineData]`.

using the `RepeatAttribute` with `[Theory]` and set it to repeat 5 times takes 528ms and using the `[Fact]` and use for loop to run 5 times takes 519ms so there is no much difference.

using the `RepeatAttribute` with `[Theory]` and set it to repeat 5 times 528ms totally:
image
image

using the `[Fact]` and use for loop to run 5 times 519ms:
image

### CODE REVIEW

File: src/Test.FunctionalTests.BinSkim.Driver/AnalyzeCommandTests.cs [hunk 1]
```diff
@@ -113,6 +113,32 @@ public void AnalyzeCommand_Hashes_ShouldUpdateDataToInsert()
...
+                Force = true,
+                Recurse = true,
+                Threads = 10,
+                IgnorePdbLoadError = false,
+                DataToInsert = new[] { OptionallyEmittedData.Hashes },
+            };
+            var command = new MultithreadedAnalyzeCommand();
+            command.Run(options);
+            var log = SarifLog.Load(fileName);
+            log.Runs[0].Invocations[0].ToolConfigurationNotifications.Should().BeNull($"iteration {iterationNumber}");
```
[the remark this answers is not in the document]
Comment (shaopeng-gh) [reply src/Test.FunctionalTests.BinSkim.Driver/AnalyzeCommandTests.cs]: done getting the new pipeline changes, looks good.
( I have to disable this test for Linux, the pdb load is for windows only. )

**Review by suvamM — COMMENTED** [review]
@shaopeng-gh The changes look good to me overall. But in the PR description, could you add where you got the test dlls from? Are they drops provided by the customer? If yes, is it OK to put them in the public repository?

**Comment by shaopeng-gh** [comment]
Yes. I have added in the PR description at the top.

---
In reply to: [1394680198](https://github.com/microsoft/binskim/pull/856#issuecomment-1394680198) [](http://example.com/codeflow?ancestors=1394680198,1394680198)

**Comment by suvamM** [comment]
> Yes. I have added in the PR description at the top.
> 
> In reply to: [1394680198](https://github.com/microsoft/binskim/pull/856#issuecomment-1394680198)

I see, understood.

**Review by suvamM — APPROVED** [review]

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Pull request merged**